apicast/src/policy.lua

```diff
@@ -0,0 +1,26 @@
+local _M = {}
+
+local setmetatable = setmetatable
+local policy_chain = require('policy_chain')
+
+local noop = function() end
+
+function _M.new(name)
+    local policy = {
+        _NAME = name or 'policy',
+        _VERSION = '0.0',
+    }
+    local mt = { __index = policy }
+
+    function policy.new()
+        return setmetatable({}, mt)
+    end
+
+    for i=1,#(policy_chain.PHASES) do
+        policy[policy_chain.PHASES[i]] = noop
+    end
+
+    return policy, mt
```

Comment:
I'd document what this method returns as I think returning the `mt` might be a bit counter-intuitive. I think we should document that it's returned so `policy.new` can be overriden. Any other reason?

Reply:
That is the only reason so far. I could also store it in `_M` like `_M.class` or `_M.mt`.  Not sure what is better though. 

But for sure has to be documented as this is public API for new policies.

Reply:
I'll try to set the `mt` as metatable of the returned module. Then you could get it by `getmetatable`. Maybe it won't cause infinite loop :)

Reply:
In the end went with not returning it at all and letting the user to decide what to do with it. They should upvalue the new function to a variable and call it later.